cmd/kube-proxy/app/server_others.go

```diff
@@ -268,8 +268,9 @@ func tryIPVSProxy(iptver iptables.IPTablesVersioner, kcompat iptables.KernelComp
 	// IPVS Proxier relies on iptables
 	useIPVSProxy, err := ipvs.CanUseIPVSProxier()
 	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("can't determine whether to use ipvs proxy, using userspace proxier: %v", err))
-		return proxyModeUserspace
+		// Try to fallback to iptables before falling back to userspace
+		utilruntime.HandleError(fmt.Errorf("can't determine whether to use ipvs proxy, error: %v, trying iptables proxier", err))
+		return tryIPTablesProxy(iptver, kcompat)
```

Comment:
If we fail to find the requirements needed to run IPVS and return here, can we remove https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-proxy/app/server_others.go#L287-L289 ?

Reply:
That's a nice finding :)

Can we remove https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-proxy/app/server_others.go#L279? We can not simply remove https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-proxy/app/server_others.go#L287-L289 since `tryIPVSProxy() string` needs a return string.